### PR TITLE
Restore compatibility with 4.02

### PR DESCRIPTION
**Pull request by nojb**
I did some brief testing on my side, but would appreciate confirmation from some other dev (especially if this is not being tested in the CI).

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]
Can you add a changelog entry ? I'm gonna merge the 2.9.2 changes update for main now.

File: CHANGES.md [hunk 1]
```diff
@@ -308,6 +308,9 @@ Unreleased
 - Switch to the stag formatting API to be compatible with OCaml 5.0 (#5350,
   @emillon).
 
+- Disable warning for deprecated Toploop functions used in dune files written in
+  OCaml syntax. Restores 4.02 compatibility. (#5381, @nojb)
+
```
Comment (ejgallego) [inline CHANGES.md:313]: That should go in a 2.9.3 release entry, if you rebase on main the 2.9.2 entry has been updated now. Sorry for annoying with this.

Comment (nojb) [reply CHANGES.md:310]: Done!

**Comment by ejgallego** [comment]
I've tested a backport of this on lwt_react.1.1.5 + ocaml-base-compiler.4.02.3 which is the failure @kit-ty-kate reported.

Unfortunately it doesn't work:
```
#=== ERROR while compiling lwt_react.1.1.5 ====================================#
# context     2.1.0 | linux/x86_64 | ocaml-base-compiler.4.02.3 | https://opam.ocaml.org#ccfbafd8
# path        ~/.opam/dev-402/.opam-switch/build/lwt_react.1.1.5
# command     ~/.opam/opam-init/hooks/sandbox.sh build dune build -p lwt_react -j 15
# exit-code   1
# env-file    ~/.opam/log/lwt_react-44273-2f38c9.env
# output-file ~/.opam/log/lwt_react-44273-2f38c9.out
### output ###
# Error: Syntax error: attributes not expected.
# [...]
# (cd src/ppx && /home/egallego/.opam/dev-402/bin/ocaml -I +compiler-libs /home/egallego/.opam/dev-402/.opam-switch/build/lwt_react.1.1.5/_build/.dune/default/src/ppx/dune.ml)
# File "/home/egallego/.opam/dev-402/.opam-switch/build/lwt_react.1.1.5/_build/.dune/default/src/ppx/dune.ml", line 27, characters 0-457:
# Error: Syntax error: attributes not expected.
#        ocaml (internal) (exit 2)
# (cd src/domain && /home/egallego/.opam/dev-402/bin/ocaml -I +compiler-libs /home/egallego/.opam/dev-402/.opam-switch/build/lwt_react.1.1.5/_build/.dune/default/src/domain/dune.ml)
# File "/home/egallego/.opam/dev-402/.opam-switch/build/lwt_react.1.1.5/_build/.dune/default/src/domain/dune.ml", line 27, characters 0-457:
# Error: Syntax error: attributes not expected.
#        ocaml (internal) (exit 2)
# (cd src/core && /home/egallego/.opam/dev-402/bin/ocaml -I +compiler-libs /home/egallego/.opam/dev-402/.opam-switch/build/lwt_react.1.1.5/_build/.dune/default/src/core/dune.ml)
# File "/home/egallego/.opam/dev-402/.opam-switch/build/lwt_react.1.1.5/_build/.dune/default/src/core/dune.ml", line 27, characters 0-457:
# Error: Syntax error: attributes not expected.
```

**Comment by nojb** [comment]
> Unfortunately it doesn't work:

OK, let me try something else.

**Comment by kit-ty-kate** [comment]
Could the tests be enabled for 4.02 ?

**Comment by nojb** [comment]
> > Unfortunately it doesn't work:
> 
> OK, let me try something else.

I pushed an alternative fix that should work on 4.02.

File: CHANGES.md (hunk 1 above)
Comment (ejgallego) [reply CHANGES.md:310]: Thanks a lot!

**Comment by nojb** [comment]
Just thinking about this again, perhaps we want to simply pass `-w -3` on the command line when executing the OCaml script (instead of disabling the warning just for `Toplevel.directive_table`)? That way we protect ourselves against any future deprecation as well, no?

**Comment by nojb** [comment]
> Just thinking about this again, perhaps we want to simply pass `-w -3` on the command line when executing the OCaml script (instead of disabling the warning just for `Toplevel.directive_table`)? That way we protect ourselves against any future deprecation as well, no?

The downside is that 1) we may miss deprecations if we are not warned about them, and 2) the deprecation alerts will also *not* be triggered for the user code.

**Pull request merged**